### PR TITLE
feat: stepaction for analyzing test results

### PR DESCRIPTION
**Pull request by psturc**
### JIRA
https://issues.redhat.com/browse/KFLUXDP-42

### Verification

Tested locally with following TR:
```
apiVersion: tekton.dev/v1beta1
kind: TaskRun
metadata:
  name: test-stepaction
spec:
  taskRef:
    name: pull-request-commenter
  params:
    - name: oci-container
      value: quay.io/psturc/oci-artifacts:test-case-failure
    - name: junit-report-name
      value: e2e-report.xml
    - name: e2e-log-name
      value: e2e-tests.log
    - name: cluster-provision-log-name
      value: cluster-provision.log
    - name: test-name
      value: testanalysis
    - name: pull-request-author
      value: psturc
    - name: pull-request-number
      value: "1"
    - name: git-repo
      value: "testyaml"
    - name: git-org
      value: "psturc"
    - name: pipeline-aggregate-status
      value: "Failed"
```

Result: https://github.com/psturc/testyaml/pull/1#issuecomment-2488939331

### CODE REVIEW

**Comment by flacatus** [comment]
@psturc I see your Pull Request example https://github.com/psturc/testyaml/pull/1#issuecomment-2488939331... It is possible to make the test suite that failed clicable and see the errors when i click in the suite?

File: common/stepactions/analyze-test-results/0.1/analyze-test-results.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      description: JUnit file report name for analysis
+    - name: e2e-log-name
+      type: string
+      default: e2e-tests.log
+      description: The name of the log from running tests
+    - name: cluster-provision-log-name
+      type: string
+      default: cluster-provision.log
+      description: The name of the log from provisioning a testing cluster
+  workingDir: $(params.workspace-path)
```
Comment (flacatus) [inline common/stepactions/analyze-test-results/0.1/analyze-test-results.yaml:34]: also we can use volumes. But up to you here

**Comment by psturc** [comment]
> @psturc I see your Pull Request example [psturc/testyaml#1 (comment)](https://github.com/psturc/testyaml/pull/1#issuecomment-2488939331)... It is possible to make the test suite that failed clicable and see the errors when i click in the suite?

@flacatus  by that you mean make the log from the test case collapsible? If so, that would be possible, but would require a change in qe-tools code.


Screenshot 2024-11-22 at 9 00 56

**Comment by flacatus** [comment]
> > @psturc I see your Pull Request example [psturc/testyaml#1 (comment)](https://github.com/psturc/testyaml/pull/1#issuecomment-2488939331)... It is possible to make the test suite that failed clicable and see the errors when i click in the suite?
> 
> @flacatus by that you mean make the log from the test case collapsible? If so, that would be possible, but would require a change in qe-tools code.
> 
> Screenshot 2024-11-22 at 9 00 56

Yes, will be nice if can be collapsable. If all fail, the comment will be super huge

**Comment by psturc** [comment]
@flacatus ah, ok, that should be covered in https://issues.redhat.com/browse/KFLUXDP-9 :)

**Comment by flacatus** [comment]
LGTM

File: common/stepactions/analyze-test-results/0.1/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# analyze-test-results stepaction
+
+This StepAction runs "analyze-test-results" command from qe-tools binary against supplied OCI artifact that contains logs and test results.
```
Comment (Dannyb48) [inline common/stepactions/analyze-test-results/0.1/README.md:3]: I'm kind of wondering if we should include a link to the cmd in the qe-tools repo? And maybe append  the same sentence "which tries to identify the cause of the failure and provide the result of the analysis"

**Review by Dannyb48 — CHANGES_REQUESTED** [review]

File: common/stepactions/analyze-test-results/0.1/README.md (hunk 2 above)
Comment (psturc) [reply common/stepactions/analyze-test-results/0.1/README.md]: thanks, updated

**Review by Dannyb48 — APPROVED** [review]

**Pull request merged**